src/App.js

```diff
@@ -1,61 +1,33 @@
-  const [playerTwoHand, setPlayerTwoHand] = useState([]);
-  const [playerThreeHand, setPlayerThreeHand] = useState([]);
-  const [from, setFrom] = useState('deck');
-  const [to, setTo] = useState(1);
-
-  function findCardIndex(value, suit, cards) {
-    return cards.findIndex(card => card.value === value && card.suit === suit);
-  }
-
-  function passCard(card) {
-    const playerHands = [playerOneHand, playerTwoHand, playerThreeHand];
-    const playerHandSetFunctions = [setPlayerOneHand, setPlayerTwoHand, setPlayerThreeHand];
-
-    // arrays start at zero, but our players start at 1 :shrug:
-    const toHand = playerHands[to - 1] || deck;
-    const fromHand = playerHands[from - 1] || deck;
-
-    const toSetFunction = playerHandSetFunctions[to - 1] || setDeck;
-    const fromSetFunction = playerHandSetFunctions[from - 1] || setDeck;
-
-    const cardToMoveIndex = findCardIndex(card.value, card.suit, fromHand);
-    const [cardToMove] = fromHand.splice(cardToMoveIndex, 1);
-
-    toHand.push(cardToMove);
-
-    toSetFunction([...toHand]);
-    fromSetFunction([...fromHand]);
-
-    setSelectedCard(null);
-  }
-
+  const {
+    selectedCard, deck, playerOneHand, playerTwoHand, playerThreeHand
+  } = useCardGameContext();
   return (
-    <div className="App">
-      <section>
-        {/* if the player names are numbers, that will make our life easier later because we can reuse numbers as arrays. Note that this will make our app brittle! */}
-        <Player to={to} player={1} hand={playerOneHand} setFrom={setFrom} selectedCard={selectedCard} setTo={setTo} setSelectedCard={setSelectedCard} />
-        <Player to={to} player={2} hand={playerTwoHand} setFrom={setFrom} selectedCard={selectedCard} setTo={setTo} setSelectedCard={setSelectedCard} />
-        <Player to={to} player={3} hand={playerThreeHand} setFrom={setFrom} selectedCard={selectedCard} setTo={setTo} setSelectedCard={setSelectedCard} />
-        <CardList cards={deck} selectedCard={selectedCard} setSelectedCard={setSelectedCard} setFrom={setFrom} player={'deck'} />
-      </section>
-      <section>
-        {
-          selectedCard && <ExecutePassButton passCard={passCard} from={from} to={to} selectedCard={selectedCard} />
-        }
-      </section>
-    </div>
+    <CardGameProvider>
+      <div className="App">
+        <header>
+          <h1>WELCOME</h1>
+        </header>
+        <section>
+          {/* if the player names are numbers, that will make our life easier later because we can reuse numbers as arrays. Note that this will make our app brittle! */}
+          <Player player={1} hand={playerOneHand} />
+          <Player player={2} hand={playerTwoHand} />
+          <Player player={3} hand={playerThreeHand} />
+          <CardList cards={deck} player={'deck'} />
+        </section>
+        <section>
+          {
+            selectedCard && <ExecutePassButton />
+          }
+        </section>
+      </div>
   );
```

Comment:
line 30 needs to appear after your closing provider tag on 31 - it looks like this is the only error your tests are failing on